keystore/v2/keystore/api/keyStore.go

```diff
@@ -34,4 +44,31 @@ type MutableKeyStore interface {
 	// OpenKeyRingRW opens a modifiable key ring with given purpose.
 	// A new key ring will be created if it does not exist yet.
 	OpenKeyRingRW(purpose string) (MutableKeyRing, error)
+
+	// ImportKeyRings unpacks key rings packaged by ExportKeyRings.
+	// The provided cryptosuite is used to verify the signature on the container and decrypt key ring data.
+	// Optional delegate can be used to control various aspects of the import process, such as conflict resolution.
+	ImportKeyRings(exportData []byte, cryptosuite *crypto.KeyStoreSuite, delegate *KeyRingImportDelegate) error
+}
+
+// ImportDecision constants describe how to proceed with import conflict resolution.
+type ImportDecision int
+
+// ImportDecision options.
+const (
+	// Do not modify existing key ring, abort with given error.
+	ImportAbort ImportDecision = iota
+	// Do not modify existing key ring, proceed with importing others.
+	ImportSkip
+	// Overwrite existing key ring with new data.
+	ImportOverwrite
+)
+
+// KeyRingImportDelegate controls details of key ring import process.
+type KeyRingImportDelegate struct {
```

Comment:
why to not use interface instead struct?

Reply:
> why to not use interface instead struct?

Hm... The idea was that there could be more than one aspect of key import process that the user would like to control, so I wanted to make it possible to specify them independently, possibly without specifying some parts they don’t care about.

However, now the users typically have to construct a `KeyRingImportDelegate` and still fill it in with some predefined strategies. Your comment made me realize that we can really just use an interface for that and provide a bunch of premade strategies for the most common use cases. If that does not fit the bill then go ahead, implement the entire `KeyRingImportDelegate` interface the way you want it.

I’ll change this to an interface instead.